### ISSUE TITLE
verify the checksum in bookie write path

### ISSUE THREAD

**Issue by sijie** [apache/bookkeeper#1046]
**FEATURE REQUEST**

1. Please describe the feature you are requesting.

Current we only verify checksum on read path, but we don't verify checksum on write path. So a corrupted entry might be written to all replicas on write path (e.g. the message is corrupted at the source).

It is required to verify checksum in bookie write path before a bookie writes it.

2. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

*must-have*

3. Provide any additional detail on your proposed use case for this feature.

It becomes achievable after #1008

**Comment by karanmehta93** [comment]
Checkout https://quip.com/FmlEAnMbtjnU
@sijie @eolivelli Feel free to comment or add new ideas and your thoughts.

**Comment by Ghatage** [comment]
@karanmehta93 the checksum in the write path is a great addition. Can you contribute it to the community code?
